pkg/agent/cniserver/pod_configuration.go

```diff
@@ -570,3 +588,175 @@ func (pc *podConfigurator) disconnectInterceptedInterface(podName, podNamespace,
+		return fmt.Errorf("interface config %s is not found in local cache", ifName)
+	}
+
+	// Update interface config with the ofPort.
+	ifConfig.OVSPortConfig.OFPort = ofPort
+	m.ifaceStore.UpdateInterface(ifConfig)
+
+	// Install OpenFlow entries for the Pod.
+	klog.V(2).InfoS("Setting up Openflow entries for OVS port", "port", ifName)
+	if err := m.ofClient.InstallPodFlows(ifName, ifConfig.IPs, ifConfig.MAC, uint32(ofPort), ifConfig.VLANID, nil); err != nil {
+		return fmt.Errorf("failed to add Openflow entries for OVS port %s: %v", ifName, err)
+	}
+
+	// Notify the Pod update event to required components.
+	event := agenttypes.PodUpdate{
+		PodName:      ifConfig.PodName,
+		PodNamespace: ifConfig.PodNamespace,
+		IsAdd:        true,
+		ContainerID:  ifConfig.ContainerID,
+	}
+	m.podUpdateNotifier.Notify(event)
+	return nil
+}
+
+func (m *podIfaceMonitor) updatePodUnreadyAnnotation(ctx context.Context, pod *corev1.Pod, addAnnotation bool) error {
+	podName, podNamespace := pod.Name, pod.Namespace
+	annotated := false
+	if pod.Annotations != nil {
+		_, annotated = pod.Annotations[agenttypes.PodNotReadyAnnotationKey]
+	}
+
+	if addAnnotation && !annotated {
+		// Add the annotation on Pod with '"pod.antrea.io/not-ready": ""'
+		patch, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"annotations": map[string]interface{}{agenttypes.PodNotReadyAnnotationKey: ""},
+			},
+		})
+		_, err := m.kubeClient.CoreV1().Pods(podNamespace).Patch(ctx, podName, apitypes.MergePatchType, patch, metav1.PatchOptions{})
+		if err != nil {
+			return err
+		}
+	} else if !addAnnotation && annotated {
+		// Remove the annotation on Pod with '"pod.antrea.io/not-ready": ""'
+		patch, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"annotations": map[string]interface{}{agenttypes.PodNotReadyAnnotationKey: nil},
+			},
+		})
+		_, err := m.kubeClient.CoreV1().Pods(podNamespace).Patch(context.Background(), podName, apitypes.MergePatchType, patch, metav1.PatchOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *podIfaceMonitor) updateUnReadyPod(status *openflow15.PortStatus) {
+	ovsPort := string(bytes.Trim(status.Desc.Name, "\x00"))
+	podInfo, found := m.unReadyInterfaces[ovsPort]
+	if !found {
+		klog.InfoS("OVS port is not found", "ovsPort", ovsPort)
+		return
+	}
+
+	ofPort := status.Desc.PortNo
+	if err := m.updatePodFlows(ovsPort, int32(ofPort)); err != nil {
+		klog.ErrorS(err, "Failed to update Pod's OpenFlow entries, requeue the PortStatus message after 10s", "PodName", podInfo.podName, "PodNamespace", podInfo.podNamespace, "OVSPort", ovsPort)
+		m.reQueuePortStatus(status)
+		return
+	}
+
+	podInfo.flowInstalled = true
+}
+
+func (m *podIfaceMonitor) checkUnReadyPods() {
+	for key, podInfo := range m.unReadyInterfaces {
+		ctx := context.Background()
+		pod, err := m.kubeClient.CoreV1().Pods(podInfo.podNamespace).Get(ctx, podInfo.podName, metav1.GetOptions{})
```

Comment:
Instead of annotating the Pod, I feel it's more user-friendly and natural to generate events for the Pod. Users and who debug this can get this exception of a Pod from single truth when they describe the Pod, instead of needing to know Antrea specific annotation.

And the event implementation supports merging similar events and rate-limiting the event generation of a Pod.

The code could just keep generating event of warning when pod is not ready, and generate event of normal when it's ready.

@antoninbas what do you think?

Reply:
Good catch, that sounds better to me.
Do we need any new RBAC permissions to generate such events for Pods?